### PR TITLE
Fix PPSDiamondDQM histogram of rechits in the last lumisection

### PR DESCRIPTION
**Pull request by ChrisMisan**
#### PR description:
PR fixes the the rechits plot which is suppose to represent the last lumisection. Before the fix plot was being filled with all the lumisections.

#### PR validation:
standard tests

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

backport to 12_4_X will follow

### CODE REVIEW

**Comment by ChrisMisan** [comment]
type bug-fix, ctpps

**Comment by ChrisMisan** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

File: DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc [hunk 1]
```diff
@@ -686,9 +686,18 @@ std::shared_ptr<dds::Cache> CTPPSDiamondDQMSource::globalBeginLuminosityBlock(co
...
   d->hitDistribution2dMap.reserve(potPlots_.size());
   if (!perLSsaving_ && plotOnline_) {
-    for (auto& plot : potPlots_)
-      d->hitDistribution2dMap[plot.first] =
-          std::unique_ptr<TH2F>(static_cast<TH2F*>(plot.second.hitDistribution2d_lumisection->getTH2F()->Clone()));
+    for (auto& plot : potPlots_) {
+      d->hitDistribution2dMap[plot.first] = std::make_unique<TH2F>(
+          "hits in planes lumisection",
+          (std::string(plot.second.hitDistribution2d_lumisection->getTH2F()->GetTitle()) + ";plane number;x (mm)")
+              .c_str(),
```
Comment (makortel) [inline DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc:693]: I believe this leads to use-after-free. The return value of `c_str()`, i.e. the pointer,  is kept alive until the `make_unique` returns, but the temporary string is not.

Comment (ChrisMisan) [reply DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc:693]: I am afraid I do not follow, of course this array is kept alive, c_str() is just returning pointer to char* which is a buffer of a string with trailing null, you can imagine there's no c_str() invocation, then you are expecting this argument to be deallocated before it is passed to a function? @makortel  can you please clarify, maybe I am not understanding this correctly.

Comment (makortel) [reply DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc:693]: Sorry for the noise, I was wrong. The temporary string is indeed kept alive until the `make_unique` returns.

**Pull request merged**